### PR TITLE
Bed rotation fix

### PR DESCRIPTION
**Pull request by Cyber-Finn**
## About the PR
I reworked the buckling and rotation systems. 
Unsure how tbh, but previous changes that someone made in the last week somehow cause the game to put players at a weird angle when they're strapped to beds (either by themselves or others).
The code looked 100% perfect though - even when debugging - but through magic and no-sleep, I have managed to resolve the issue.

## Why / Balance
No balance changes made.
This was changed because the current code/game causes players to be placed at a weird angle, which breaks immersion. This is just a basic QoL update, to fix player rotation when on beds.

## Technical details
I reworked the system that handles buckling and strapping players to objects like beds, chairs, etc.

## Media
Here's a video of my fix at work:
https://github.com/space-wizards/space-station-14/assets/146375585/91e8b221-fd02-49de-b29a-6205e9c09858

Here's a video of the issue currently in prod [Specifically the LRP Leviathan server]:
https://github.com/space-wizards/space-station-14/assets/146375585/5dddc165-7c7c-4449-8c37-f3bc87a22771

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
No breaking changes

**Changelog**

:cl:
- fix: I reworked the buckling system, so players are no longer strapped to beds at a weird angle. All is right with the world.

### CODE REVIEW

**Comment by K-Dynamic** [comment]
How does this also affect player rotation and buckling on vehicles?

**Comment by Cyber-Finn** [comment]
> Does this also affect player rotation and buckling on vehicles?

good question! the change does affect the vehicle/chair aspects of the buckling system, but they still operate as normal/before.
In the video I shared on the PR, you can see me strap into a chair, and it aligns me the right way - it's the same for vehicles (they apparently count as chairs lol)

File: Content.Shared/Rotation/RotationVisualsComponent.cs [hunk 1]
```diff
@@ -9,14 +9,16 @@ public sealed partial class RotationVisualsComponent : Component
     /// <summary>
     /// Default value of <see cref="HorizontalRotation"/>
     /// </summary>
+    ///
+    //should we not be changing the default rotation to be either horizontal or vertical?
     [DataField]
-    public Angle DefaultRotation = Angle.FromDegrees(90);
+    public Angle DefaultRotation = Angle.Zero; //should this not be 0 degrees? as the character stands up by default (as opposed to 90 degrees - laying down)
```
Comment (ElectroJr) [inline Content.Shared/Rotation/RotationVisualsComponent.cs:15]: AFAIK the variable is just badly named, its meant to be the default rotation an entity should have when horizontal. I think it just came about as a result of some buckling shitcode to stop dead people from being horizontal while buckled to a chair. So AFAIK it is meant to be 90.
@AC

File: Content.Shared/Rotation/RotationVisualsComponent.cs [hunk 2]
```diff
@@ -9,14 +9,16 @@ public sealed partial class RotationVisualsComponent : Component
...
 
     [DataField]
-    public Angle VerticalRotation = 0;
+    public Angle VerticalRotation = Angle.Zero;
 
     [DataField, AutoNetworkedField]
-    public Angle HorizontalRotation = Angle.FromDegrees(90);
+    public Angle HorizontalRotation = Angle.FromDegrees(180); //this should be -90, because the beds are all at rotation -90, regular 90 degrees will have you facing the wrong way
```
Comment (ElectroJr) [inline Content.Shared/Rotation/RotationVisualsComponent.cs:21]: Why was this changed to 180? dead mobs shouldn't default to upside down.

**Comment by AJCM-git** [comment]
@Cyber-Finn you coming back to this?

**Comment by VasilisThePikachu** [comment]
Superseded by #24746
